modules/gdscript/gdscript.cpp

```diff
@@ -872,11 +872,8 @@ Error GDScript::reload(bool p_keep_state) {
 		if (source_path.is_empty()) {
 			source_path = get_path();
 		}
-		if (!source_path.is_empty()) {
-			MutexLock lock(GDScriptCache::singleton->lock);
```

Comment:
Not sure about the removal of this lock. `shallow_gdscript_cache` seems to be accessed within a lock always.

Reply:
Oh! This is a leftover of a previous refactor. Thanks, I'll reinstate the lock.